src/orion/benchmark/task/profet/profet_task.py

```diff
@@ -0,0 +1,817 @@
+
+        # def seeded_sample(_, n_samples: int = 1, seed=None):
+        #     nonlocal self
+        #     if seed is None:
+        #         seed = self._np_rng
+        #     return _original_sample(n_samples=n_samples, seed=seed)
+
+        # self._space.sample = seeded_sample
+
+        self.name = f"profet.{type(self).__qualname__.lower()}_{self.task_id}"
+
+    @contextmanager
+    def seed_randomness(self):
+        """Context manager used to seed the portions of the code that use randomness, without
+        affecting the global state.
+        """
+        devices = [] if self.device.type == "cpu" else [self.device]
+        if self._np_rng is None:
+            # Initialize the numpy random state.
+            self._np_rng = RandomState(seed=self.seed)
+
+        start_np_state = np.random.get_state()
+        # Temporarily set the global random state to that of our RNG.
+        np.random.set_state(self._np_rng.get_state())
+
+        # Need to do this when it comes to the torch random state, since `torch.distributions` uses
+        # the global state when sampling, and there are no ways of passing a seed to a distribution.
+        with torch.random.fork_rng(devices=devices):
+            if self._torch_rng_state is None:
+                torch.random.manual_seed(self.seed)
+            else:
+                torch.random.set_rng_state(self._torch_rng_state)
+
+            # Yield, which probably executes a few lines with randomness.
+            yield
+
+            # Update the torch random state.
+            self._torch_rng_state = torch.random.get_rng_state()
+
+        # NOTE: Re-creating a RandomState object and setting the state, since I don't know how to
+        # get the 'global' RandomState object.
+        self._np_rng = RandomState(seed=self.seed)
+        self._np_rng.set_state(np.random.get_state())
+
+        np.random.set_state(start_np_state)
+
+    def call(
+        self,
+        x: Union[InputType, Trial, Dict, Tuple] = None,
+        with_grad: bool = False,
+        **kwargs,
+    ) -> List[Dict]:
+        """Get the value of the sampled objective function at the given point (hyper-parameters).
+
+        If `with_grad` is passed, also returns the gradient of the objective function with respect
+        to the inputs.
+
+        Parameters
+        ----------
+        x : Union[InputType, Trial, Dict, Tuple]
+            Either a Trial, a dataclass, a dict, or a tuple, that is a point from this tasks' space.
+        with_grad : bool, optional
+            Wether to also compute the gradients with respect to the inputs, by default False.
+
+        Returns
+        -------
+        List[Dict]
+            Result dictionaries: objective and optionally gradient.
+
+        Raises
+        ------
+        ValueError
+            If the input isn't of a supported type.
+        """
+        logger.debug(f"received x={x}")
+        if x is None and kwargs:
+            x = kwargs
+
+        if is_dataclass(x):
```

Comment:
I think we had a discussion on the topic but I cannot find it. Why would it support different formats for x? Aren't they supposed to be suggestions from the algorithms which are all following the same convention?

Reply:
Testing like for unit-testing? I honestly do not see why we would want to test the method with other types than what it would be supposed to handle. Supporting other types does not prevent us from supporting typical point format but it does look like feature creep which isn't great either. 

When you say _their dataclasses_, you mean something like an instance of `FcNetTaskHParams`? Soon enough we will drop the point array format and use trial objects instead. At that point the benchmarks should follow this transition and support Trials as the input of the call methods. The call method of the tasks is intended to be following the same interface as the functions users would pass to experiment_client.workon() so we need to be careful not to create inconsistencies.

Reply:
Fixed: Made the input type for each task into a `TypedDict`, (dictionary with type annotations), rather than dataclasses, and the `call` of `ProfetTask` now only supports a single type of input.

Reply:
Also customized the `__call__` so that Benchmark calling `task(**hparam_dict)` won't cause an issue.

Reply:
It would be simpler to define `def call(**x)` no? It is a bit weird/inconsistent otherwise, because Oríon expects the functions that are optimized to take the parameters as arguments, not a dict of the parameters.

Reply:
I'd currently argue that it isn't. I can explain why in our next 1-1 meeting.

Reply:
Ok. We should write back a summary of our discussion here for traceability.

Reply:
I'll try to summarize here. @lebrice Please correct me if I say anything wrong of forget something. So the issue is that the **kwargs input would causing issue with typing linters because child class would define a different interface than **kwargs. I reread the code and I am a bit confused because I don't see child class overwritting the call method.

An alternative I would see that could be more `typing-friendly` is to rename `call` to something like `execute` and then in each subclass write `call(with, the, proper, arguments)` and convert them into a dict passed to `execute`. It does not add value in terms of features, but it makes the code a bit more explicit.

On the same topic, tasks are expected to rewrite `call`, not `__call__`.

Reply:
Ok, fixed in c3a384643b396a399ad04f1b2eae9650fa89950e

Reply:
Thanks! The docstring of basetask.call would need to be updated. There are no x argument. Could you remove the *args as well? The method should never be called without the argument names.

Reply:
Does a0b16047b8a027f0f9ff068504770007829fbc0c match what you'd expect?

Reply:
Looks good, thanks!